drivers/gpio/gpio_pca95xx.c

```diff
@@ -0,0 +1,671 @@
+/*
+ * Copyright (c) 2015 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file Driver for PCA95XX I2C-based GPIO driver.
+ */
+
+#include <errno.h>
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <sys/util.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+
+#include "gpio_pca95xx.h"
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(gpio_pca95xx);
+
+/* Register definitions */
+#define REG_INPUT_PORT0			0x00
+#define REG_INPUT_PORT1			0x01
+#define REG_OUTPUT_PORT0		0x02
+#define REG_OUTPUT_PORT1		0x03
+#define REG_POL_INV_PORT0		0x04
+#define REG_POL_INV_PORT1		0x05
+#define REG_CONF_PORT0			0x06
+#define REG_CONG_PORT1			0x07
+#define REG_OUT_DRV_STRENGTH_PORT0_L	0x40
+#define REG_OUT_DRV_STRENGTH_PORT0_H	0x41
+#define REG_OUT_DRV_STRENGTH_PORT1_L	0x42
+#define REG_OUT_DRV_STRENGTH_PORT1_H	0x43
+#define REG_INPUT_LATCH_PORT0		0x44
+#define REG_INPUT_LATCH_PORT1		0x45
+#define REG_PUD_EN_PORT0		0x46
+#define REG_PUD_EN_PORT1		0x47
+#define REG_PUD_SEL_PORT0		0x48
+#define REG_PUD_SEL_PORT1		0x49
+#define REG_INT_MASK_PORT0		0x4A
+#define REG_INT_MASK_PORT1		0x4B
+#define REG_INT_STATUS_PORT0		0x4C
+#define REG_INT_STATUS_PORT1		0x4D
+#define REG_OUTPUT_PORT_CONF		0x4F
+
+/* Driver flags */
+#define PCA_HAS_PUD			BIT(0)
+
+/**
+ * @brief Check to see if a I2C master is identified for communication.
```

Comment:
This really shouldn't be necessary.  The only way to get a device pointer is for initialization to have succeeded, in which case `i2c_master` would have been found.

Reply:
Seems like the idea behind the function is to make sure that none of the public API functions will attempt to communicate over i2c bus if the initialization of the i2c master device has failed. But the name of the function is rather confusing.

Reply:
The function is removed.